### PR TITLE
Update Threadpool metric collector to use reflection to read queue capacity

### PR DESCRIPTION
**Pull request by rguo-aws**
*Fixes #, if available:*

*Description of changes:*
threadpool capacity is currently being read from ThreadPoolStats which requires core ES changes. What we can do instead is to update Threadpool metric collector to use reflection to read queue capacity. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/ThreadPoolMetricsCollector.java [hunk 1]
```diff
@@ -186,8 +168,8 @@ public ThreadPoolStatus() {
             this.rejected = -1;
             this.threadsCount = -1;
             this.threadsActive = -1;
-            this.queueLatency = null;
-            this.queueCapacity = null;
+            this.queueLatency = -1.0;
+            this.queueCapacity = -1;
```
Comment (sruti1312) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/ThreadPoolMetricsCollector.java:172]: We might need to set to the queueCapacity value right?

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/ThreadPoolMetricsCollector.java [hunk 2]
```diff
@@ -75,33 +79,27 @@ public void collectMetrics(long startTime) {
...
+                    Object queue = threadPoolExecutor.getQueue();
+                    // TODO: we might want to read the capacity of SifiResizableBlockingQueue in the future.
+                    // In order to do that we can create a new PerformanceAnalyzerLibrary package and push
+                    // all the code which depends on core ES specific changes into that library.
+                    if (queue instanceof SizeBlockingQueue) {
+                        return ((SizeBlockingQueue)queue).capacity();
+                    }
                 } catch (Exception e) {
-                    latency = null;
+                    LOG.warn("Fail to read queue capacity via reflection");
```
Comment (sruti1312) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/ThreadPoolMetricsCollector.java:95]: Add e to the message?

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/ThreadPoolMetricsCollector.java (hunk 1 above)
Comment (rguo-aws) [reply src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/ThreadPoolMetricsCollector.java:172]: this is the default setter for unit test. so we set all fields in this class to -1

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/ThreadPoolMetricsCollector.java [hunk 3]
```diff
@@ -142,32 +140,16 @@ public long getRejected() {
         public final long rejected;
         public final int threadsCount;
         public final int threadsActive;
-        @JsonInclude(Include.NON_NULL)
-        public final Double queueLatency;
-        @JsonInclude(Include.NON_NULL)
-        public final Integer queueCapacity;
-
-          public ThreadPoolStatus(String type,
```
Comment (khushbr) [inline src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/ThreadPoolMetricsCollector.java:150]: Is this constructor no longer used ?  I am assuming we were using this for testing purpose.

**Review by khushbr — APPROVED** [review]

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/ThreadPoolMetricsCollector.java (hunk 1 above)
Comment (sruti1312) [reply src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/ThreadPoolMetricsCollector.java:172]: Got it

**Review by sruti1312 — APPROVED** [review]

File: src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/ThreadPoolMetricsCollector.java (hunk 3 above)
Comment (rguo-aws) [reply src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/ThreadPoolMetricsCollector.java:150]: I checked the codebase and this constructor is not used anywhere.

**Pull request merged**